src/AWSS3.jl

```diff
@@ -58,6 +58,12 @@ const SSDict = Dict{String,String}
 const AbstractS3Version = Union{AbstractString,Nothing}
 const AbstractS3PathConfig = Union{AbstractAWSConfig,Nothing}
 
+# Utility function to workaround https://github.com/JuliaCloud/AWS.jl/issues/547
+function get_robust_case(x, key)
+    haskey(x, key) && return x[key]
+    return x[lowercase(key)]
```

Comment:
I think we should switch the order, so the error makes more sense if there isn't a key at all (regardless of case). I.e. the KeyError should probably have the standard casing instead of the lowercase casing:
```suggestion
    lkey = lowercase(key)
    haskey(x, lkey) && return x[lkey]
    return x[key]
```